presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/MetadataResolver.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.common.CatalogSchemaName;
+import com.facebook.presto.common.QualifiedObjectName;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.MaterializedViewDefinition;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Metadata resolver provides information about catalog, schema, tables, views, types, and functions required for analyzer functionality.
+ */
+public interface MetadataResolver
+{
+    /**
+     * Returns if the catalog exists in the metadata for a given catalog name.
+     */
+    boolean catalogExists(String catalogName);
+
+    /**
+     * Returns if the catalog and schema exist in the metadata for a given catalog and schema name.
+     */
+    boolean schemaExists(CatalogSchemaName schemaName);
+
+    /**
+     * Returns if the table, schema and catalog exist in the metadata for a given catalog, schema and table name.
+     */
+    boolean tableExists(QualifiedObjectName tableName);
+
+    /**
+     * Returns the list of column metadata for the provided catalog, schema and table name.
+     * This method throws exception if the provided table is not present in the metadata.
+     */
+    Optional<List<ColumnMetadata>> getColumns(QualifiedObjectName tableName);
+
+    /**
+     * Returns view metadata for a given catalog, schema and view name.
+     */
+    Optional<ViewDefinition> getView(QualifiedObjectName viewName);
+
+    /**
+     * Returns if provided catalog, schema and object name is a view.
+     */
+    default boolean isView(QualifiedObjectName viewName)
+    {
+        return getView(viewName).isPresent();
+    }
+
+    /**
+     * Returns materialized view metadata for a given catalog, schema and materialized view name
```

Comment:
Returns the materialized view metadata for a named view

Reply:
"named view" sounds confusing to me.